code/modules/atmospherics/machinery/pipes/multiz.dm

```diff
@@ -43,15 +40,14 @@
 	center.pixel_x = PIPING_LAYER_P_X * (piping_layer - PIPING_LAYER_DEFAULT)
 	. += center
 
-
 ///Attempts to locate a multiz pipe that's above us, if it finds one it merges us into its pipenet
 /obj/machinery/atmospherics/pipe/multiz/pipeline_expansion()
 	var/turf/T = get_turf(src)
-	var/obj/machinery/atmospherics/pipe/multiz/above = locate(/obj/machinery/atmospherics/pipe/multiz) in(SSmapping.get_turf_above(T))
-	var/obj/machinery/atmospherics/pipe/multiz/below = locate(/obj/machinery/atmospherics/pipe/multiz) in(SSmapping.get_turf_below(T))
-	if(below)
-		below.pipeline_expansion() //If we've got one below us, force it to add us on facebook
-	if(above)
-		nodes += above
-		above.nodes += src //Two way travel :)
+	for(var/obj/machinery/atmospherics/pipe/multiz/above in SSmapping.get_turf_above(T))
+		if(above && above.piping_layer == piping_layer)
+			nodes += above
+			above.nodes += src //Two way travel :)
+	for(var/obj/machinery/atmospherics/pipe/multiz/below in SSmapping.get_turf_below(T))
+		if(below && below.piping_layer == piping_layer)
```

Comment:
The checks for if(whatever) aren't needed here, since nulls will get filtered by the for loop

Reply:
removed